ioc_finder/ioc_finder.py

```diff
@@ -418,100 +419,133 @@ def find_iocs(  # noqa: CCR001 pylint: disable=R0912,R0915
     original_text = text
 
     # urls
-    iocs['urls'] = parse_urls(text, parse_urls_without_scheme=parse_urls_without_scheme)
-    if not parse_domain_from_url and not parse_from_url_path:
-        text = _remove_items(iocs['urls'], text)
-    elif not parse_domain_from_url:
-        text = _percent_decode_url(iocs['urls'], text)
-        text = _remove_url_domain_name(iocs['urls'], text)
-    elif not parse_from_url_path:
-        text = _percent_decode_url(iocs['urls'], text)
-        text = _remove_url_paths(iocs['urls'], text)
-    else:
-        text = _percent_decode_url(iocs['urls'], text)
+    if len(data_types) == 0 or "urls" in data_types:
```

Comment:
There are multiple instances in which we are testing `len(data_types) == 0`. I recommend that we modify this as follows:

1. Define a constant that is a list of all possible indicator types
2. Set the default argument for `data_types` to the constant defined in step 1
3. Remove the references to `len(data_types) == 0`

In this way, if no `data_types` are specified, we assume we want to parse everything.

So the pseudo-code would look something like:

```python
...
POSSIBLE_DATA_TYPES = ['urls', ...]
...

def find_iocs(..., data_types: List[str] = POSSIBLE_DATA_TYPES, ...):
    ...

    if 'urls' in data_types:
        ...

    if 'xmpp_addresses' in data_types:
        ...

    ...
```

Reply:
Never done this type of coding, but if I understand correctly this should be an easy fix. Thanks!

Reply:
Ok done - if you could do the testing for me, that would be great. I just made it run +1:

Reply:
Thanks for updating that! And sure, I'll update the tests.